examples/nlp/gpt/train_gpt_sft.py

```diff
@@ -209,6 +204,22 @@ def main(cfg) -> None:
         load_gbs=True,
     )
 
+    # nemo uses the train dataloader to figure out
+    # max steps to take when max_steps = -1
+    # but our train dataloader is for the prompts
+    # so we instaniate a dummy dataloader
+    # to get the proper max *optimization* steps
+    # nemo treats batch size of normal dataloader as GBS/DP
+    # so we need to offset it by DP
+    dummy_train_dataloader = torch.utils.data.DataLoader(
+        dataset=train_ds, batch_size=divide(cfg.model.global_batch_size, parallel_state.get_data_parallel_world_size())
+    )
+
+    init_using_ptl(trainer, ptl_model, dummy_train_dataloader, train_ds)
+    # make sure the dummy train dataloader is never used
+    del ptl_model._train_dl
+    del dummy_train_dataloader
```

Comment:
This needs some clean-up -- the comments are copy/pasted from PPO which is a very different setting than SFT => why do you need this here? Also is it intended to have two calls to `init_using_ptl`?

I'm not actually sure what issue you're trying to fix, but just in case it might be related, note that I finally got this merged into NeMo: https://github.com/NVIDIA/NeMo/pull/7518

Reply:
I copy-pasted this following @gshennvm's suggestion on how to fix incorrect LR schedule. If there is a better way to do this, I'd be happy to.

The second call to init_using_ptl is a bug, thanks for catching this - I have a separate copy of the starting script that I was using to test this and forgot to move that change over!

Reply:
So the code you copy/pasted is motivated in the PPO case because the true train dataloader essentially counts the number of generation steps instead of the number of optimization steps, so we need to create a fake one to make the scheduler based off the number of optimization steps.

I guess the main question is what do you mean by "incorrect LR schedule" for SFT? If it's the problem described in the NeMo PR I linked to above, this should be the proper fix.

Reply:
The incorrect schedule was just something I observed - it was moving much slower than expected (barely started decay after the full epoch when training finished). I guess it might have been fixed by your PR, let me double check this